qcodes/dataset/data_set.py

```diff
@@ -436,14 +428,9 @@ def add_metadata(self, tag: str, metadata: Any):
         Args:
             tag: represents the key in the metadata dictionary
             metadata: actual metadata
-
         """
-        # TODO: this follows the spec but another option:
-        # json_meta_data = json.dumps(metadata)
-        # add_meta_data(self.conn, self.run_id, {"metadata": json_meta_data})
-
         add_meta_data(self.conn, self.run_id, {tag: metadata})
-        # adding meta-data does not commit
+        # `add_meta_data` does not commit, hence we commit here:
```

Comment:
Perhaps better to make `add_meta_data` do an atomic transaction?

Reply:
this will require a bit more refactoring because it will most probably touch `update_meta_data` and `insert_meta_data` function which i frankly did not want to worry about in this PR.